proposed/activity-improvements.md

```diff
@@ -0,0 +1,290 @@
+        public override int GetHashCode()
+    }
+```
+
+ ### ActivityLink
+```C#
+namespace System.Diagnostics
+{
+    /// <summary>
+    /// Activity may be linked to zero or more other <see cref="ActivityContext"/> that are causally related.
+    /// Links can point to ActivityContexts inside a single Trace or across different Traces.
+    /// Links can be used to represent batched operations where a Activity was initiated by multiple initiating Activities,
+    /// each representing a single incoming item being processed in the batch.
+    public readonly struct ActivityLink
+    {
+        public ActivityLink(ActivityContext context) 
+        public ActivityLink(ActivityContext context, IDictionary<string, object>? attributes)
+        public ActivityContext Context { get; }
+        public IDictionary<string, object>? Attributes { get; }
+    }
+}
+```
+
+### Activity
+
+```C#
+namespace System.Diagnostics
+{
+    public partial class Activity : IDisposable // added IDisposable 
+    {
+        ....
+        public System.Diagnostics.ActivityKind Kind { get; set; }
+        public System.Collections.Generic.IEnumerable<ActivityLink> Links { get; }
+        public System.Diagnostics.Activity AddLink(ActivityLink link) 
+
+        public static System.Diagnostics.Activity? StartNew(string name) 
+        public static System.Diagnostics.Activity? StartNew(string name, ActivityContext context, IEnumerable<ActivityLink>? links = null, DateTimeOffset statTime = default) 
+        public static System.Diagnostics.Activity? StartNew(string name, ActivityContext context, ActivityKind kind, IEnumerable<System.Collections.Generic.KeyValuePair<string,string>>? tags = null, IEnumerable<ActivityLink>? links = null, DateTimeOffset startTime = default) 
+
+        public bool TryGetContext(out ActivityContext context)
+
+        public static IDisposable StartListening(
+                                                 Func<string, ActivityContext, ActivityKind, IEnumerable<System.Collections.Generic.KeyValuePair<string,string>>?, IEnumerable<ActivityLink>?, bool> shouldCreateActivity, 
+                                                 Action<Activity> onActivityStarted, 
+                                                 Action<Activity> onActivityStopped);
+
+        public void SetCustomProperty(string propertyName, object? propertyValue)
+        public object? GetCustomProperty(string propertyName) 
+
+        public void Dispose() 
+    }
+}
+
+```
+
+Q & A 
+================
+
+[TODO] There is a bunch of discussion in https://github.com/dotnet/runtime/issues/31373 that hasn't been replicated here. If those discussions continue it is probably worthwhile to put conclusions (with needed context here)
+
+
+
+Q\) It would be great if we put some guidance here regarding the Activity naming convention
+
+[Noah] Agreed. It wasn't an area I focused on much in that experiment, but we will need to have a plan for it.
+
+
+
```

Comment:
New Q&A issue needed. Moving from conversation from [the old issue](https://github.com/dotnet/runtime/issues/31373#issuecomment-600908953)

@SergeyKanzhelev wrote:

> IsRecording vs. no activity allocation. I commented before that not creating activity breaks propagation of the context. Which will break user scenarios. So we need Activity to be created or at least context propagated. Thus it needs some indication on whether recording enabled or not.

Reply:
I'm not sure I understand how the OT propagation scenario is supposed to work. The API for creating a span says it only takes the parent span context implying that the newly created span has a new unique SpanId. However when IsRecording is false the new span isn't reported to a SpanProcessor which suggests that nothing has logged the parent-child relationship. The trace-id and tracestate would get propagated through but the span id chain is now broken. Is that considered the proper propagation and what we are trying to achieve here?